imgee/tasks.py

```diff
@@ -1,35 +1,85 @@
+import re
 import redis
-from imgee import app
+
+
+class InvalidRedisQueryException(Exception):
+    pass
 
 
 class TaskRegistry(object):
-    def __init__(self, name='default', connection=None):
-        self.connection = redis.from_url(connection) if connection else None
+    def __init__(self, name='default', connection=None, app=None):
         self.name = name
-        self.key = 'imgee:registry:%s' % name
+        self.connection = connection
+        self.key_prefix = 'imgee:registry:%s' % name
+        self.filename_pattern = '^[a-z0-9\_\.]+$'
```

Comment:
making it a raw string and not escaping `_`. And we dont need to compile because as you mentioned before, no point compiling during runtime. passing the pattern directly to `re.match()`.